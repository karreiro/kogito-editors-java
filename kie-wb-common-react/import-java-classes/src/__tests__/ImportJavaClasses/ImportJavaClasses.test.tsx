/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { fireEvent, render } from "@testing-library/react";
import { ImportJavaClasses } from "../../components";
import * as _ from "lodash";

describe("ImportJavaClasses component tests", () => {

    test("should render ImportJavaClasses Button component", () => {
        const { container } = render(<ImportJavaClasses buttonDisabledStatus={false} />);

        expect(container).toMatchSnapshot();
    });

    test("Should show Modal after clicking on the button", () => {
        const { baseElement, getByText } = render(<ImportJavaClasses buttonDisabledStatus={false} />);
        const modalWizardButton = getByText("Import Java classes")! as HTMLButtonElement;
        modalWizardButton.click();

        expect(baseElement).toMatchSnapshot();
    });

    test("Should search box works", () => {
      const mockedLSPGetClassService = jest.fn();
      lspGetClassServiceMock(mockedLSPGetClassService);
      const { baseElement, getByText } = render(<ImportJavaClasses buttonDisabledStatus={false} />);
      const modalWizardButton = getByText("Import Java classes")! as HTMLButtonElement;
      modalWizardButton.click();
      const inputElement = baseElement.querySelector('[aria-label="Search input"]')! as HTMLInputElement;
      expect(inputElement).toHaveValue("");
      expect(baseElement.querySelector('[aria-label="Reset"]')! as HTMLButtonElement).not.toBeInTheDocument();
      fireEvent.change(inputElement, { target: { value: "test" } });
      expect(inputElement).toHaveValue("test");
      const resetButton = baseElement.querySelector('[aria-label="Reset"]')! as HTMLButtonElement;
      expect(resetButton).toBeInTheDocument();
      resetButton.click();
      expect(baseElement.querySelector('[aria-label="Reset"]')! as HTMLButtonElement).not.toBeInTheDocument();
    });

    test("Should close Modal after opening it and clicking on the Cancel button", () => {
        const { baseElement, getByText } = render(<ImportJavaClasses buttonDisabledStatus={false} />);
        const modalWizardButton = getByText("Import Java classes")! as HTMLButtonElement;
        modalWizardButton.click();
        const cancelButton = getByText("Cancel") as HTMLButtonElement;
        cancelButton.click();

        expect(baseElement).toMatchSnapshot();
    });

  function lspGetClassServiceMock(mockedBroadcastDefinition: jest.Mock) {
    window.envelopeMock = _.extend(window.envelopeMock  || {}, {
      lspGetClassServiceMocked: (value :string) =>
        mockedBroadcastDefinition(value),
    });
  }

});
