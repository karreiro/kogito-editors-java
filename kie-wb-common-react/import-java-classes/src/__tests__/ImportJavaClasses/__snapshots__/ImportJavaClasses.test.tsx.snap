// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportJavaClasses component tests Should show Modal after clicking on the button 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div
    aria-hidden="true"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-secondary"
      data-ouia-component-id="OUIA-Generated-Button-secondary-2"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      data-testid="modal-wizard-button"
      type="button"
    >
      Import Java classes
    </button>
  </div>
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-describedby="pf-wizard-description-1"
          aria-labelledby="pf-wizard-title-1"
          aria-modal="true"
          class="pf-c-modal-box pf-m-lg"
          data-ouia-component-id="OUIA-Generated-Modal-large-2"
          data-ouia-component-type="PF4/ModalContent"
          data-ouia-safe="true"
          id="pf-modal-part-1"
          role="dialog"
        >
          
          <div
            class="pf-c-wizard"
          >
            <div
              class="pf-c-wizard__header"
            >
              <button
                aria-disabled="false"
                aria-label="Close"
                class="pf-c-button pf-m-plain pf-c-wizard__close"
                data-ouia-component-id="OUIA-Generated-Button-plain-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  />
                </svg>
              </button>
              <h2
                aria-label="Import Java classes"
                class="pf-c-title pf-m-3xl pf-c-wizard__title"
                id="pf-wizard-title-1"
              >
                Import Java classes
              </h2>
              <p
                class="pf-c-wizard__description"
                id="pf-wizard-description-1"
              >
                It converts your java assets to DMN type. This is a one-time import action: if the Java class get updated, you will need to reimport it
              </p>
            </div>
            <button
              aria-expanded="false"
              aria-label="Wizard Toggle"
              class="pf-c-wizard__toggle"
            >
              <ol
                class="pf-c-wizard__toggle-list"
              >
                <li
                  class="pf-c-wizard__toggle-list-item"
                >
                  <span
                    class="pf-c-wizard__toggle-num"
                  >
                    1
                  </span>
                   
                  Select Java classes
                </li>
              </ol>
              <span
                class="pf-c-wizard__toggle-icon"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 320 512"
                  width="1em"
                >
                  <path
                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="pf-c-wizard__outer-wrap"
            >
              <div
                class="pf-c-wizard__inner-wrap"
              >
                <nav
                  aria-labelledby="pf-wizard-title-1"
                  class="pf-c-wizard__nav"
                >
                  <ol
                    class="pf-c-wizard__nav-list"
                  >
                    <li
                      class="pf-c-wizard__nav-item"
                    >
                      <button
                        aria-current="page"
                        aria-disabled="true"
                        class="pf-c-wizard__nav-link pf-m-current pf-m-disabled"
                        disabled=""
                      >
                        Select Java classes
                      </button>
                    </li>
                    <li
                      class="pf-c-wizard__nav-item"
                    >
                      <button
                        aria-current="false"
                        aria-disabled="true"
                        class="pf-c-wizard__nav-link pf-m-disabled"
                        disabled=""
                      >
                        Cherry-pick fields
                      </button>
                    </li>
                    <li
                      class="pf-c-wizard__nav-item"
                    >
                      <button
                        aria-current="false"
                        aria-disabled="true"
                        class="pf-c-wizard__nav-link pf-m-disabled"
                        disabled=""
                      >
                        Review
                      </button>
                    </li>
                  </ol>
                </nav>
                <main
                  aria-labelledby="pf-wizard-title-1"
                  class="pf-c-wizard__main"
                >
                  <div
                    class="pf-c-wizard__main-body"
                  >
                    <p>
                      Step 1 content
                    </p>
                  </div>
                </main>
              </div>
              <footer
                class="pf-c-wizard__footer"
              >
                <button
                  aria-disabled="true"
                  class="pf-c-button pf-m-primary pf-m-disabled"
                  data-ouia-component-id="OUIA-Generated-Button-primary-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="submit"
                >
                  Next
                </button>
                <div
                  class="pf-c-wizard__footer-cancel"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-link"
                    data-ouia-component-id="OUIA-Generated-Button-link-1"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ImportJavaClasses component tests Should show Modal after clicking on the button 2`] = `
<body
  class=""
>
  <div>
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-secondary"
      data-ouia-component-id="OUIA-Generated-Button-secondary-3"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      data-testid="modal-wizard-button"
      type="button"
    >
      Import Java classes
    </button>
  </div>
  <div />
</body>
`;

exports[`ImportJavaClasses component tests should render ImportJavaClasses Button component 1`] = `
<div>
  <button
    aria-disabled="false"
    class="pf-c-button pf-m-secondary"
    data-ouia-component-id="OUIA-Generated-Button-secondary-1"
    data-ouia-component-type="PF4/Button"
    data-ouia-safe="true"
    data-testid="modal-wizard-button"
    type="button"
  >
    Import Java classes
  </button>
</div>
`;
